--- xen.spec	2014-06-16 01:32:24.045769059 +0800
+++ xen_with_spice_enabled.spec	2014-06-16 01:17:50.910240599 +0800
@@ -12,6 +12,8 @@
 %define build_efi 0
 %define with_sysv 1
 %define with_systemd 0
+%define enable_spice 1
+%define disable_sdl 1
 
 # Hypervisor ABI
 %define hv_abi  4.2
@@ -19,7 +21,7 @@
 Summary: Xen is a virtual machine monitor
 Name:    xen
 Version: 4.2.4
-Release: 30%{?dist}
+Release: 100%{?dist}
 Group:   Development/Libraries
 License: GPLv2+ and LGPLv2+ and BSD
 URL:     http://xen.org/
@@ -87,7 +89,10 @@
 Patch1005: xen-centos-blktap25-ctl-ipc-restart.patch
 
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root
-BuildRequires: transfig libidn-devel zlib-devel texi2html SDL-devel curl-devel
+BuildRequires: transfig libidn-devel zlib-devel texi2html curl-devel
+%if !%disable_sdl
+BuildRequires: SDL-devel
+%endif
 BuildRequires: libX11-devel python-devel ghostscript texlive-latex
 BuildRequires: ncurses-devel gtk2-devel libaio-devel libtool
 # for the docs
@@ -140,6 +145,10 @@
 BuildRequires: mingw64-binutils
 %endif
 
+%if %enable_spice
+BuildRequires: spice-server-devel
+%endif
+
 %description
 This package contains the XenD daemon and xm command line
 tools, needed to manage virtual machines running under the
@@ -274,6 +283,16 @@
 pushd `pwd`
 cd ${RPM_BUILD_DIR}/%{name}-%{version}/tools/qemu-xen
 #%patch105 -p1
+ 
+%if %enable_spice
+#sed -i 's/--enable-xen/& --enable-spice --enable-usb-redir/' ../Makefile
+sed -i 's/--enable-xen/& --enable-spice/' ../Makefile
+%endif
+
+%if %disable_sdl
+sed -i 's/--enable-xen/& --disable-sdl/' ../Makefile
+%endif
+
 popd
 
 # stubdom sources
